cdap-app-templates/cdap-program-report/src/test/java/co/cask/cdap/report/ReportGenerationAppTest.java

```diff
@@ -93,13 +95,32 @@ public void testGenerateReport() throws Exception {
     // Will remove once the TMS subscriber writing to the run meta fileset is implemented.
     DataSetManager<FileSet> fileSet = getDataset(metaFileset);
     populateMetaFiles(fileSet.get().getBaseLocation());
-    URL avroUrl =
-      ClassLoaders.getClassPathURL("com.databricks.spark.avro.DefaultSource",
-                                   getClass().getClassLoader().getResource(
-                                     "com/databricks/spark/avro/DefaultSource.class"));
-    File jarFile = new File(avroUrl.toURI());
-    ApplicationManager app =
-      deployApplication(ReportGenerationApp.class, jarFile);
+
+    // Trace the dependencies for the spark avro
+    ApplicationBundler bundler = new ApplicationBundler(new ClassAcceptor() {
+      @Override
+      public boolean accept(String className, URL classUrl, URL classPathUrl) {
+        if (className.startsWith("org.apache.spark.")) {
+          return false;
+        }
+        if (className.startsWith("scala.")) {
+          return false;
+        }
+        if (className.startsWith("org.apache.hadoop.")) {
+          return false;
+        }
+        if (className.startsWith("org.codehaus.janino.")) {
+          return false;
+        }
+        return true;
+      }
+    });
+
+    Location avroSparkBundle = Locations.toLocation(TEMP_FOLDER.newFile());
+    bundler.createBundle(avroSparkBundle, DefaultSource.class);
```

Comment:
can you explain/comment on why we are using `DefaultSource` class ?

Reply:
It is the class from Spark-avro, which we need to include it and its dependencies into the application jar. They are not used directly in the application code, hence can't be found via dependency tracing.